src/Domain/CoordinateMaps/FocallyLiftedEndcap.cpp

```diff
@@ -0,0 +1,331 @@
+  get<1, 0>(inv_jacobian_matrix) = get<0, 1>(inv_jacobian_matrix);
+  // dybar/dy
+  get<1, 1>(inv_jacobian_matrix) =
+      one_over_r_q - square(ybar) * jacobian_factor;
+
+  return inv_jacobian_matrix;
+}
+
+boost::optional<std::array<double, 3>> Endcap::inverse(
+    const std::array<double, 3>& target_coords,
+    const double sigma_in) const noexcept {
+  const double x = target_coords[0] - center_[0];
+  const double y = target_coords[1] - center_[1];
+  const double z = target_coords[2] - center_[2];
+
+  // Are we in the range of the map?
+  // The equal_within_roundoff below has an implicit scale of 1,
+  // so the inverse may fail if radius_ is very small on purpose,
+  // e.g. if we really want a tiny tiny domain for some reason.
+  const double r = sqrt(square(x) + square(y) + square(z));
+  if (not equal_within_roundoff(r, radius_)) {
+    return boost::none;
+  }
+
+  // Compute zbar and check if we are in the range of the map.
+  const double zbar = 2.0 * sigma_in - 1.0;
+  if (abs(zbar) > 1.0 and not equal_within_roundoff(abs(zbar), 1.0)) {
+    return boost::none;
+  }
+
+  const double rho = sqrt(square(x) + square(y));
+  if (UNLIKELY(rho == 0.0)) {
+    // If x and y are zero, so are xbar and ybar,
+    // so we are done.
+    return std::array<double, 3>{{0.0, 0.0, zbar}};
+  }
+
+  // Note: theta_max_ cannot be zero for a nonsingular map.
+  const double rhobar = atan2(rho, z) / theta_max_;
+
+  // Check if we are outside the range of the map.
+  if (rhobar > 1.0 and not equal_within_roundoff(rhobar, 1.0)) {
+    return boost::none;
+  }
+
+  const double xbar = x * rhobar / rho;
+  const double ybar = y * rhobar / rho;
+  return std::array<double, 3>{{xbar, ybar, zbar}};
+}
+
+template <typename T>
+tt::remove_cvref_wrap_t<T> Endcap::sigma(
+    const std::array<T, 3>& source_coords) const noexcept {
+  return 0.5 * (source_coords[2] + 1.0);
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 3> Endcap::deriv_sigma(
+    const std::array<T, 3>& source_coords) const noexcept {
+  using ReturnType = tt::remove_cvref_wrap_t<T>;
+  return std::array<ReturnType, 3>{
+      {make_with_value<ReturnType>(dereference_wrapper(source_coords[0]), 0.0),
+       make_with_value<ReturnType>(dereference_wrapper(source_coords[0]), 0.0),
+       make_with_value<ReturnType>(dereference_wrapper(source_coords[0]),
+                                    0.5)}};
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 3> Endcap::dxbar_dsigma(
+    const std::array<T, 3>& source_coords) const noexcept {
+  using ReturnType = tt::remove_cvref_wrap_t<T>;
+  return std::array<ReturnType, 3>{
+      {make_with_value<ReturnType>(dereference_wrapper(source_coords[0]), 0.0),
+       make_with_value<ReturnType>(dereference_wrapper(source_coords[0]), 0.0),
+       make_with_value<ReturnType>(dereference_wrapper(source_coords[0]),
+                                    2.0)}};
+}
+
+boost::optional<double> Endcap::lambda_tilde(
```

Comment:
[picking random spot] I agree with @geoffrey4444 that having the order of the functions in the documentation and in the code be consistent would make the code easier to review and understand